data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorConfig.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import org.opensearch.dataprepper.plugins.processor.configuration.KeysConfig;
+import org.opensearch.dataprepper.plugins.processor.configuration.ServiceTypeOptions;
+import org.opensearch.dataprepper.plugins.processor.configuration.AwsAuthenticationOptions;
+
+import java.util.List;
+
+/**
+ * An implementation class of GeoIP Processor configuration
+ */
+public class GeoIPProcessorConfig {
+
+    @JsonProperty("aws")
+    @NotNull
+    @Valid
+    private AwsAuthenticationOptions awsAuthenticationOptions;
```

Comment:
What will AWS configurations be used for in this plugin? I think we can remove these.

Reply:
We would need this for Connecting to S3 for the mmdb file download. So this is required.

Reply:
@ashoktelukuntla , We are moving away from this for S3 downloads. It will be provided by `data-prepper-config.yaml`.

Reply:
We have agreed to keep it in Pipelines and move it later ?